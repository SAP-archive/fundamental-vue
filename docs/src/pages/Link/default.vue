<title>Link</title>

<template>
  <div>
    <h3>Normal</h3>
    <p>
      Lorem ipsum
      <FdLink href="#" @click="handleClick">dolor sit amet consectetur</FdLink>
      adipiscing elit.
    </p>

    <h3>Selected</h3>
    <p>
      Lorem ipsum
      <FdLink href="#" selected @click="handleClick"
        >dolor sit amet consectetur</FdLink
      >
      adipiscing elit.
    </p>

    <h3>Disabled</h3>
    <p>
      Lorem ipsum
      <FdLink href="#" disabled @click="handleClick"
        >dolor sit amet consectetur</FdLink
      >
      adipiscing elit.
    </p>
  </div>
</template>
<script>
export default {
  methods: {
    handleClick() {
      console.log("clicked");
    }
  }
};
</script>
<style scoped>
p {
  margin-bottom: 20px;
}
</style>
