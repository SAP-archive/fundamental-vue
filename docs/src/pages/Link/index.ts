import { ExampleCollectionFunction } from "../types";

export const plugin: ExampleCollectionFunction = () => {
  return { status: "stable", icon: "chain-link", related: ["FdLink"] };
};
