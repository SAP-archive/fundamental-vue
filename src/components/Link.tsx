import {
  Component,
  Prop,
} from 'vue-property-decorator';
import { Api } from '@/api';
import { componentName } from '@/util';
import TsxComponent from '@/vue-tsx';

interface Props {
  selected?: boolean;
  disabled?: boolean;
}

@Component({ name: componentName('Link') })
@Api.Component('Link')
@Api.Event('click', 'Sent when link was clicked')
@Api.defaultSlot('Link Title')
export class Link extends TsxComponent<Props> {
  @Api.Prop('whether link is selected', prop => prop.type(Boolean))
  @Prop({ type: Boolean, default: false, required: false })
  public selected!: boolean;

  @Api.Prop('whether link is disabled', prop => prop.type(Boolean))
  @Prop({ type: Boolean, default: false, required: false })
  public disabled!: boolean;

  public render() {
    const attributes = {
      attrs: this.$attrs,
      on: this.$listeners,
    };
    return (
      <a
        class={this.classes}
        {...attributes}
      >
        {this.$slots.default}
      </a>
    );
  }

  private get classes() {
    return {
      'fd-link': true,
      'is-selected': this.selected,
      'is-disabled': this.disabled,
    };
  }
}
