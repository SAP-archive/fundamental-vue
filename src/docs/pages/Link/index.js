export default {
  relatedComponents: ["fd-link"]
};
