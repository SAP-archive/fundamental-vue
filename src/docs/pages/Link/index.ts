import { ExampleCollectionFunction } from '../types';

export const plugin: ExampleCollectionFunction = ({ Link }) => {
  return { icon: 'chain-link', relatedComponents: [Link] };
};
