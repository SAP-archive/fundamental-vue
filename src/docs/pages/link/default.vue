<template>
  <div>
    <FdTile>
      <FdTileContent>
        <p>
          This is how
          <fd-link href="#" data-bla="yyy" @click="handleClick">links look like by default</fd-link
          >. Isn't that nice?
        </p>
      </FdTileContent>
    </FdTile>

    <FdTile>
      <FdTileContent>
        <p>
          This is how
          <fd-link href="#" selected @click="handleClick">selected links look like by</fd-link>.
          Isn't that nice?
        </p>
      </FdTileContent>
    </FdTile>

    <FdTile>
      <FdTileContent>
        <p>
          This is how
          <fd-link href="#" disabled @click="handleClick">disabled links look like by</fd-link>.
          Isn't that nice?
        </p>
      </FdTileContent>
    </FdTile>
  </div>
</template>
<script>
export default {
  methods: {
    handleClick() {
      console.log('clicked') // eslint-disable-line no-console
    }
  }
}
</script>
