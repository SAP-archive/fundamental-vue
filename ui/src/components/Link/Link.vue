<template>
  <a v-on="listeners" @click="click" :class="classes" href="#" :title="title">
    <slot>{{ title }}</slot>
  </a>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "FdLink",
  props: {
    title: String,
    selected: { type: Boolean, default: false },
    disabled: { type: Boolean, default: false }
  },
  methods: {
    click(event: Event): void {
      if (event.defaultPrevented) {
        return;
      }
      if (this.disabled) {
        event.preventDefault();
        event.stopPropagation();
        return;
      }
      this.$emit("click", event);
    }
  },
  computed: {
    listeners(): object {
      const { click, ...listeners } = this.$listeners;
      return listeners;
    },
    classes(): object {
      return {
        "fd-link": true,
        "is-selected": this.selected,
        "is-disabled": this.disabled
      };
    }
  }
});
</script>
