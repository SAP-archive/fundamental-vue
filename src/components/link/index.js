import Link from './link.vue'
import pluginify from './../../util/pluginify'
export default pluginify(Link)
export { Link }
