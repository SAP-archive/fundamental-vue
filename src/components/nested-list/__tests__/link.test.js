import { createLocalVue, mount } from '@vue/test-utils'
import Link from './../link.vue'

describe('nested-link', () => {
  it('renders href', () => {
    const wrapper = mount(Link, {
      attrs: {
        href: '#hello'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
