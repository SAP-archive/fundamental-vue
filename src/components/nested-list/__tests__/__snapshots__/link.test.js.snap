// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nested-link renders href 1`] = `
<a
  aria-expanded="false"
  aria-haspopup="false"
  class="fd-nested-list__link"
  href="#hello"
/>
`;
